registry/remote/repository.go

```diff
@@ -1080,10 +1087,15 @@ func (s *manifestStore) push(ctx context.Context, expected ocispec.Descriptor, c
 	}
 	defer resp.Body.Close()
 
-	if resp.StatusCode != http.StatusCreated {
+	switch resp.StatusCode {
+	case http.StatusCreated:
+		return verifyContentDigest(resp, expected.Digest)
+	case http.StatusBadRequest:
+		err := errutil.ParseErrorResponse(resp)
+		return fmt.Errorf("%s: %s: %w: %v", expected.Digest, expected.MediaType, ErrBadRequest, err)
```

Comment:
It is weird to have `ErrBadRequest` here as it cannot indicate invalid manifest.

That is, `MANIFEST_INVALID` implies but is not implied by `400 Bad Request`.

Reply:
Furthermore, the `ErrBadRequest` does not contain the context. In other words, we are not able to distinguish a real bad request with an unsupported manifest in a `oras.Copy()` scenario with lots of manifests.